src/main/host/descriptor/socket.c

```diff
@@ -400,6 +405,11 @@ Packet* legacysocket_removeFromInputBuffer(LegacySocket* socket, const Host* hos
         if(socket->inputBufferLength <= 0) {
             legacyfile_adjustStatus((LegacyFile*)socket, STATUS_FILE_READABLE, FALSE);
         }
+        /* the size of input buffer changes, so we flip the flag */
+        if (length > 0) {
+            gboolean parity = (legacyfile_getStatus((LegacyFile*)socket) & STATUS_FILE_INPUT_BUFFER_PARITY) != 0;
+            legacyfile_adjustStatus((LegacyFile*)socket, STATUS_FILE_INPUT_BUFFER_PARITY, !parity);
+        }
```

Comment:
Is this true? For example if you read from a socket, will epoll always trigger an edge-triggered "readable" event? This is something that there should be a test for.

Reply:
Good point. Not sure either. I will check that again.